sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/OfferQueue.java

```diff
@@ -0,0 +1,94 @@
+package com.mesosphere.sdk.scheduler;
+
+import org.apache.mesos.Protos;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.stream.Collectors;
+
+/**
+ * This class acts as a buffer of Offers from Mesos.  By default it holds a maximum of 100 Offers.
+ */
+public class OfferQueue {
+    public static final int DEFAULT_CAPACITY = 100;
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final BlockingQueue<Protos.Offer> queue;
+
+    public OfferQueue () {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public OfferQueue(int capacity) {
+        this.queue = new LinkedBlockingQueue<>(capacity);
+    }
+
+    /**
+     * Calling this method is a blocking Operation. It returns all Offers currently in the queue. It always returns at
+     * least one Offer.
+     */
+    public List<Protos.Offer> takeAll() {
+        List<Protos.Offer> offers = new LinkedList<>();
+        try {
+            offers.add(queue.take());
```

Comment:
Should add a comment here describing the blocking behavior/why both calls are needed